File: python/largest-series-product/largest_series_product.py
import math

def largest_product(series, size):
    if size < 0:
        raise ValueError("span must not be negative")
    if size > len(series):
        raise ValueError("span must be smaller than string length")
    if series == "" or size == 0:
        return 1

    digits = to_digits(series)
    return max([math.prod(digits[i:i+size]) for i in range(len(digits)-size+1)])

def to_digits(series):
    try:
        return [int(a) for a in series]
    except ValueError:
        raise ValueError("digits input must only contain digits")

File: csharp/largest-series-product/LargestSeriesProduct.cs
using System;
using System.Linq;

public static class LargestSeriesProduct
{
    public static long GetLargestProduct(string digits, int span) 
    {
        if (span < 1) throw new ArgumentException("span must be positive");
        if (span > digits.Length) throw new ArgumentException("span must be smaller than string length");
        if (digits.Any(ch => !char.IsDigit(ch)))
            throw new ArgumentException("invalid character found");
        
        var intDigits = digits.Select(ch => ch - '0').ToArray();
        var result = 0L;
        for (var i = 0; i < intDigits.Length - span + 1; i++)
        {
            var product = 1L;
            for (var j = 0; j < span; j++)
            {
                product *= intDigits[i + j];
            }
            result = Math.Max(product, result);
        }

        return result;
    }
}
